node/network/protocol/src/lib.rs

```diff
@@ -336,6 +337,40 @@ pub mod v1 {
 		Statement(Hash, SignedFullStatement)
 	}
 
+	#[derive(Debug, Clone, Copy, PartialEq, thiserror::Error)]
+	#[allow(missing_docs)]
+	pub enum CompressedPoVError {
+		#[error("Failed to compress a PoV")]
+		Compress,
+		#[error("Failed to uncompress a PoV")]
+		Uncompress,
+		#[error("Failed to decode the uncompressed PoV")]
+		Decode,
+	}
+
+	/// SCALE and GZip encoded [`PoV`].
+	#[derive(Debug, Clone, Encode, Decode, PartialEq)]
+	pub struct CompressedPoV(Vec<u8>);
+
+	impl CompressedPoV {
+		/// Create from the given [`PoV`].
+		pub fn from_pov(pov: &PoV) -> Result<Self, CompressedPoVError> {
```

Comment:
**Nit** (i.e. feel free to ignore): I can't help but suggest to name those: compress and uncompress